torch/_tensor_str.py

```diff
@@ -228,9 +258,14 @@ def _tensor_str(self, indent):
     summarize = self.numel() > PRINT_OPTS.threshold
     if self.dtype is torch.float16 or self.dtype is torch.bfloat16:
         self = self.float()
-    formatter = _Formatter(get_summarized_data(self) if summarize else self)
-    return _tensor_str_with_formatter(self, indent, formatter, summarize)
 
+    if self.dtype.is_complex:
+        real_formatter = _Formatter(get_summarized_data(self.real) if summarize else self.real)
+        imag_formatter = _Formatter(get_summarized_data(self.imag) if summarize else self.imag)
```

Comment:
How do the tests trigger summarize True/False?

If summarize is False then real_formatter/imag_formatter = self.real/self.imag. These are tensors which are passed to _complex_tensor_str_with_formatter(), which can call `_scalar_str(self.real, real_formatter)`, which seems like a nonsense call when real_formatter is a tensor?

Reply:
I am not sure what you are talking about. `real_formatter` and `imag_formatter` are formatters at all places.

Reply:
also I have already added the test for summarize as mentioned above :D

Reply:
resolved https://github.com/pytorch/pytorch/pull/40513#discussion_r447752639